XbSymbolDatabase.h

```diff
@@ -551,6 +951,73 @@ void XbSymbolContinuousSigScan(bool enable);
 /// <param name="enable">Input boolean to use first symbol address only or not.</param>
 void XbSymbolFirstDetectAddressOnly(bool enable);
 
+// ******************************************************************
+// * API v3 (context) functions to use with other projects.
+// ******************************************************************
+// NOTE: If need to make customize setup, please use the above functions
+// before call XbSymbol_Init function.
+
+/// <summary>
+/// Step 1: Generate library array for LibraryHeader input.
+/// First call with <paramref name="library_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="library_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="library_out">Input null pointer will provide the count to allocate memory for library_out.filters field. Otherwise, information will be input to library_out.filters field.</param>
+unsigned int XbSymbolGenerate_LibraryFilter(const void* xb_header_addr, LibraryHeader* library_out);
+
+/// <summary>
+/// Step 2: Generate section array for SectionHeader input.
+/// First call with <paramref name="section_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="section_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="section_out">Input null pointer will provide the count to allocate memory, then allocate memory to section_out.filters field.</param>
+/// <param name="is_raw">True: Convert array to use raw xbe's relative address; False: Convert array to use xbox virtual memory relative address.</param>
+unsigned int XbSymbolGenerate_SectionFilter(const void* xb_header_addr, SectionHeader* section_out, bool is_raw);
+
+/// <summary>
+/// Step 3: Create XbSymbolContext context for the scan process.
+/// </summary>
+/// <param name="ppContext">Output XbSymbolContext handler.</param>
+/// <param name="register_func">Register callback function for any detected symbols during scan process.</param>
+/// <param name="library_input">See XbSymbolGenerate_LibraryFilter for details.</param>
+/// <param name="section_input">See XbSymbolGenerate_SectionFilter for details.</param>
+/// <param name="kernel_thunk">Input decoded virtual kernel thunk.</param>
+bool XbSymbol_Init(XbSymbolContext* ppContext, xb_symbol_register_t register_func, LibraryHeader library_input, SectionHeader section_input, xbaddr kernel_thunk);
+
+/// <summary>
+/// Step 4: Perform a manual scan to set xreferences manually by requirement.
+/// </summary>
+/// <param name="pContext">Input XbSymbolContext handler.</param>
+void XbSymbolScan_Manual(XbSymbolContext pContext);
```

Comment:
These new prefixes are weird?
Functions with different prefix interact; but unlike COM, there don't seem to be objects involved.

I'd expect this to be `XbSymbolContext_ScanManual` (`Object_Method`) or `XbSymbolScanManual` (`PrefixMethod` - existing style) or `XbSymbol_ScanManual` ('Module_Method')

Reply:
True, I will fix it later on in this pull request.

Reply:
fixed